### ISSUE TITLE
Editor - long press on tab to go to dir - enhancement 

### ISSUE THREAD

**Issue by Phuket2** [omz/Pythonista-Issues#183]
It would be great if you could do a long press on a tab in the editor and be able to navigate to the directory that file is located in.  I realise you don't have a similar mechanism that does that now.  But if you don't want to do that, maybe showing the full path.  But has to be full path, not a truncated version. Another idea might be to show it in the file dropdown menu.  
Personally I have this problem a lot.  Many tabs open after days and days, or have opened a file in the editor as a result of a search script I have etc.... but it is disconnected in a way

**Comment by zrzka** [comment]
Not ideal, but you can use favorites in 311014. Long press is used to reorganize tabs. Closing this one and if anyone will come up with better idea, new issue can be filled.

**Comment by omz** [comment]
Reopening this because I think it's a reasonable feature request. Something like "Show in Library" for editor tabs would be useful, though I'm not sure yet where to put it... Favorites don't really address this problem because you'd have to be in the right folder already to add a favorite.

**Comment by zrzka** [comment]
@omz what about if revealed library always switches folder to the dirname of file path in a tab? It has it flaws, but it's better probably?

* Let's say I open `a/a.py` and whenever I reveal library, I'm in `a` folder
* Let's say I open `a/a.py`, reveal library, I'm in `a`, open `b/b.py` file, if selected tab is `b/b.py`, library points to `b`, if I select tab `a/a.py`, library points to `a` even if I did open `b/b.py` as a last file

And then library can contain some button which pops to root if I'm burried in some deep folder.

**Comment by omz** [comment]
@zrzka I think that would be seriously confusing/frustrating in a lot of cases.

**Comment by omz** [comment]
Maybe it could look like this:

![img_39de247f41f3-1](https://user-images.githubusercontent.com/82469/31929513-ad4830d2-b89c-11e7-81a1-a3e248d49766.jpeg)

Tapping the "Folder" button would open the folder that contains the current file...

**Comment by zrzka** [comment]
@omz yeah, you're right probably. It's based on what I do for example. Editing project files, viewing other files and what I need mostly is to open files that are "around" files I have in my tabs. Anyway, I think that this issue exists because there's no open quickly, no notion of project, ... It's like PyCharm where I open the whole Work folder (tens of repos) on my Mac.

**Comment by zrzka** [comment]
@omz nice one, can work :)

**Comment by omz** [comment]
An alternative would be offering a "Reveal in Library" option when long-pressing the "hamburger" icon. Not very discoverable, but makes sense semantically, and this is probably a power user feature anyway.

**Comment by zrzka** [comment]
Personally, I don't like long-pressing gestures. It's slow. Or at least, it feels it's slow. I do always prefer two taps compared to long-press for example.
